state/validation.go

```diff
@@ -62,12 +62,6 @@ func validateBlock(evidencePool EvidencePool, stateDB dbm.DB, state State, block
 			block.ConsensusHash,
 		)
 	}
-	if !bytes.Equal(block.LastResultsHash, state.LastResultsHash) {
```

Comment:
thinking about this change, I suggest we do another dedicate PR to remove those checks and make this PR only about `CheckBlock`